src/current/v23.1/changefeed-messages.md

```diff
@@ -266,7 +266,7 @@ As an example, you run the following sequence of SQL statements to create a chan
     {"after": {"id": 4, "name": "Danny", "office": "los angeles"}, "key": [4], "updated": "1701102561022789676.0000000000"}
     ~~~
 
-    The messages received at the sink are in order by timestamp **for each key**. Here, the update for key `[1]` is emitted before the insertion of key `[2]` even though the timestamp for the update to key `[1]` is higher. That is, if you follow the sequence of updates for a particular key at the sink, they will be in the correct timestamp order. However, re-emitted messages (duplicates) may not be in the correct timestamp order. For details on when changefeeds might re-emit messages, refer to [Duplicate messages](#duplicate-messages).
+    The messages received at the sink are in order by timestamp **for each key**. Here, the update for key `[1]` is emitted before the insertion of key `[2]` even though the timestamp for the update to key `[1]` is higher. That is, if you follow the sequence of updates for a particular key at the sink, they will be in the correct timestamp order. However, re-emitted messages (duplicates) may not contain all row updates after the changefeed checkpoints. For details on when changefeeds might re-emit messages, refer to [Duplicate messages](#duplicate-messages).
```

Comment:
Is this edit ok? Andy had used the "closed timestamp" in a suggestion — I switched in the changefeed "checkpoints" since we already use that language when talking about duplicates on this page. But, is this the same, i.e., the changefeed won't checkpoint until the closed timestamp has passed?

Reply:
While true, "may not contain all row updates after the changefeed checkpoints" isn't entirely accurate, either. When we re-emit messages, we start at some update that may be several updates after the last checkpoint. What we want to make clear is that the updates between that first re-emitted update and any new updates may be missing.

Reply:
@rharding6373 Maybe I'm not understanding well, but do we want to simply say:

 "re-emitted messages after the last checkpoint may not contain all row updates." 

Or, more specific: "However, if a changefeed starts to re-emit messages at some point after a checkpoint, it may not re-emit all updates between the first emitted duplicate and new updates to the table."

When you say "any new updates", you mean completely new changes to the table?

Reply:
OK @awoods187 @rharding6373 I took another edit at this sentence — let me know what you think.